src/Parser/Druid/Restoration/Modules/Talents/Photosynthesis.js

```diff
@@ -49,17 +70,80 @@ class Photosynthesis extends Analyzer {
     const spellId = event.ability.guid;
     const amount = event.amount + (event.absorbed || 0);
 
-    if(spellId === SPELLS.REJUVENATION.id && this.selectedCombatant.hasBuff(SPELLS.LIFEBLOOM_HOT_HEAL.id, null, 0, 0, this.selectedCombatant.sourceID)) {
-      this.rejuvenationIncrease += calculateEffectiveHealing(event, PHOTOSYNTHESIS_REJUV_INCREASE);
+    // Lifebloom random bloom procc
+    if(spellId === SPELLS.LIFEBLOOM_BLOOM_HEAL.id && (this.lastRealBloomTimestamp === null || (event.timestamp - this.lastRealBloomTimestamp) < BLOOM_BUFFER_MS)){
```

Comment:
The old logic was incorrect, showing stuff like this: 
![img](https://i.gyazo.com/acd6dc191aaf53758aba20e726eba41f.png)
I'll go back and double check this

Reply:
Ah yes, the old version is definitely giving wrong results. In that example there should be 0 random blooms, because Lifebloom was never cast on another player.
I suspect that would be caused if the bloom heal appeared in the log before the removebuff or refreshbuff event, or was delayed too long after those events. It might be useful to directly check for false positives by seeing if the apparently random bloom happened on the activeCombatant.